### PR TITLE
Add workflow to check for circular deps (3.0 only)

### PR DESCRIPTION
**Pull request by anphel31**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
Add a github workflow to verify changes will not introduce a circular dependency into the full builds. Circular dependencies are difficult to catch during code review, and our buddybuild pipeline does not catch them (since it only builds with a small graph, including the packages specifically requested in that build). We typically catch circular dependency breaks in nightly full builds.
This PR check is currently only enabled for 3.0 branches, as it relies on the DAILY_BUILD_ID repos.

###### Change Log  

- Add workflow to check for circular dependency loops

###### Does this affect the toolchain?  

**NO**

###### Test Methodology

- Pipeline build id: This PR has a link to the check: https://github.com/microsoft/azurelinux/actions/runs/8397479624/job/23000849407?pr=8528
- Example of the PR check failing (with intentional circular dependency added to this PR): https://github.com/microsoft/azurelinux/actions/runs/8397600261/job/23001167684?pr=8528

### CODE REVIEW

**Review by rikenm1 — APPROVED** [review]
LGTM

**Review by mbykhovtsev-ms — APPROVED** [review]

**Review by bfjelds — APPROVED** [review]

**Review by dmcilvaney — APPROVED** [review]

File: .github/workflows/check-circular-deps.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        uses: actions/checkout@v4
+
+      - name: Check for circular dependencies
+        run: |
+          echo "Checking for circular dependency loops..."
+          pushd toolkit
+          ./scripts/setuplkgtoolchain.sh
+          BUILD_ID=$(wget -qO - https://mariner3dailydevrepo.blob.core.windows.net/lkg/lkg-3.0-dev.json | jq -r ".dailybuildid" | tr '\.' '-')
+          sudo make -j$(nproc) graph REBUILD_TOOLS=y DAILY_BUILD_ID=${BUILD_ID}
+          popd
```
Comment (PawelWMS) [inline .github/workflows/check-circular-deps.yml:29]:
```suggestion
          toolkit/scripts/setuplkgtoolchain.sh
          BUILD_ID=$(wget -qO - https://mariner3dailydevrepo.blob.core.windows.net/lkg/lkg-3.0-dev.json | jq -r ".dailybuildid" | tr '\.' '-')
          sudo make -C toolkit -j$(nproc) graph REBUILD_TOOLS=y DAILY_BUILD_ID=${BUILD_ID}
```

File: .github/workflows/check-circular-deps.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      - name: Workflow trigger checkout
+        uses: actions/checkout@v4
+
+      - name: Check for circular dependencies
+        run: |
+          echo "Checking for circular dependency loops..."
+          pushd toolkit
+          ./scripts/setuplkgtoolchain.sh
+          BUILD_ID=$(wget -qO - https://mariner3dailydevrepo.blob.core.windows.net/lkg/lkg-3.0-dev.json | jq -r ".dailybuildid" | tr '\.' '-')
+          sudo make -j$(nproc) graph REBUILD_TOOLS=y DAILY_BUILD_ID=${BUILD_ID}
```
Comment (PawelWMS) [inline .github/workflows/check-circular-deps.yml:28]: Could we only add a comment to explain these steps?

**Review by PawelWMS — APPROVED** [review]

File: .github/workflows/check-circular-deps.yml (hunk 1 above)
Comment (anphel31) [reply .github/workflows/check-circular-deps.yml]: updated

File: .github/workflows/check-circular-deps.yml (hunk 2 above)
Comment (anphel31) [reply .github/workflows/check-circular-deps.yml]: Sure, updated

File: .github/workflows/check-circular-deps.yml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        run: |
+          echo "Checking for circular dependency loops..."
+          # Call this script to sync the toolchain manifests with the LKG daily build.
+          ./toolkit/scripts/setuplkgtoolchain.sh
+          # Determine the LKG daily build ID.
+          LKG_BUILD_ID=$(wget -qO - https://mariner3dailydevrepo.blob.core.windows.net/lkg/lkg-3.0-dev.json | jq -r ".dailybuildid" | tr '\.' '-')
+          # Setup the toolchain using the LKG daily build, and then make the full package graph.
+          # This will fail if any circular dependency loops are detected in the core SPECs.
+          sudo make -C toolkit -j$(nproc) graph REBUILD_TOOLS=y DAILY_BUILD_ID=${LKG_BUILD_ID}
+        if: always()
```
Comment (PawelWMS) [inline .github/workflows/check-circular-deps.yml:32]: Just curious: is this even needed?

**Review by PawelWMS — APPROVED** [review]

File: .github/workflows/check-circular-deps.yml (hunk 3 above)
Comment (anphel31) [reply .github/workflows/check-circular-deps.yml:32]: It's probably not necessary in this case. Will remove

**Pull request merged**